ui/components/app/modals/export-private-key-modal/export-private-key-modal.component.js

```diff
@@ -142,86 +178,128 @@ export default class ExportPrivateKeyModal extends Component {
               hideModal();
             }}
           >
-            {this.context.t('cancel')}
+            {t('cancel')}
           </Button>
         )}
         {privateKey ? (
           <Button
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.LG}
+            width={BLOCK_SIZES.FULL}
             onClick={() => {
-              hideModal();
+              hideModalFunc();
             }}
-            type="primary"
-            large
-            className="export-private-key-modal__button"
           >
-            {this.context.t('done')}
+            {t('done')}
           </Button>
         ) : (
           <Button
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.LG}
+            width={BLOCK_SIZES.HALF}
             onClick={() => {
-              this.context.trackEvent({
+              trackEvent({
                 category: MetaMetricsEventCategory.Keys,
                 event: MetaMetricsEventName.KeyExportRequested,
                 properties: {
                   key_type: MetaMetricsEventKeyType.Pkey,
                 },
               });
 
-              this.exportAccountAndGetPrivateKey(this.state.password, address);
+              exportAccountAndGetPrivateKey(password, address);
             }}
-            type="primary"
-            large
-            className="export-private-key-modal__button"
-            disabled={!this.state.password}
+            disabled={!password}
           >
-            {this.context.t('confirm')}
+            {t('confirm')}
           </Button>
         )}
-      </div>
+      </Box>
     );
-  }
+  };
 
-  render() {
-    const {
-      selectedIdentity,
-      warning,
-      showAccountDetailModal,
-      hideModal,
-      previousModalState,
-    } = this.props;
-    const { name, address } = selectedIdentity;
+  const { name, address } = selectedIdentity;
+
+  return (
+    <AccountModalContainer
+      className="export-private-key-modal"
+      selectedIdentity={selectedIdentity}
+      showBackButton={previousModalState === 'ACCOUNT_DETAILS'}
+      backButtonAction={() => showAccountDetailModal()}
+    >
+      {showHoldToReveal ? (
+        <HoldToRevealModal
+          onLongPressed={() => setShowHoldToReveal(false)}
+          willHide={false}
+        />
+      ) : (
+        <>
```

Comment:
I feel like this whole section should be a named functional Component

Reply:
sure, i've refactored it